### PR TITLE
feat: listing positions

### PR DESCRIPTION
**Pull request by bilalcorbacioglu**
## Summary

The mockup data was removed, and completed contract interactions.

### CODE REVIEW

File: src/state/pwallet/concentratedLiquidity/hooks/evm.ts [hunk 1]
```diff
@@ -94,17 +95,46 @@ export function useGetUserPositions() {
...
+        return [position.token0, position.token1];
+      });
+
+      const uniqueTokens = [...new Set(tokens.flat())];
+      return uniqueTokens;
+    }
+    return [];
+  }, [positions]);
+
+  const uniqueTokensWithData = useTokens(uniqueTokens);
```
Comment (SarjuHansaliya) [inline src/state/pwallet/concentratedLiquidity/hooks/evm.ts:118]: there is some type error here, please run `./node_modules/.bin/tsc` to get more details.

File: src/components/ConcentratedLiquidity/PositionCard/index.tsx [hunk 2]
```diff
@@ -12,13 +16,90 @@ import {
...
+  // if token1 is an ETH-/BTC-stable asset, set it as the base token
+  // const bases = [...Object.values(WRAPPED_NATIVE_CURRENCY), WBTC];
+  // if (bases.some((base) => base && base.equals(token1))) {
+  //   return {
+  //     priceLower: position.token0PriceUpper.invert(),
+  //     priceUpper: position.token0PriceLower.invert(),
+  //     quote: token0,
+  //     base: token1,
+  //   };
+  // }
```
Comment (SarjuHansaliya) [inline src/components/ConcentratedLiquidity/PositionCard/index.tsx:58]: do we need this comment ? if not lets remove it

File: src/components/ConcentratedLiquidity/PositionCard/index.tsx [hunk 3]
```diff
@@ -12,13 +16,90 @@ import {
   OptionButton,
   OptionsWrapper,
   Panel,
-  Price,
+  PriceWrapper,
   Row,
 } from './styles';
 import { PositionCardProps } from './types';
 
+export function getPriceOrderingFromPositionForUI(position?: Position): {
```
Comment (SarjuHansaliya) [inline src/components/ConcentratedLiquidity/PositionCard/index.tsx:24]: lets move this function to `src/components/ConcentratedLiquidity/PositionCard/helpers.tsx`

File: src/state/pwallet/concentratedLiquidity/types.ts [hunk 4]
```diff
@@ -1,21 +1,39 @@
...
+
+export interface UseV3PositionResults {
+  loading: boolean;
+  position: PositionDetails | undefined;
+}
+
+export interface UseV3PositionsResults {
+  loading: boolean;
+  positions: PositionDetailsFromChain[] | undefined;
+}
```
[the remark this answers is not in the document]
Comment (bilalcorbacioglu) [reply src/state/pwallet/concentratedLiquidity/types.ts:13]: Done.

File: src/utils/index.ts [hunk 5]
```diff
@@ -590,3 +594,27 @@ export function decimalToFraction(number: number): Fraction {
 export function capitalizeWord(word = '') {
   return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
 }
+
+export function formatTickPrice({
```
[the remark this answers is not in the document]
Comment (bilalcorbacioglu) [reply src/utils/index.ts:598]: The tick term is special to Conc Liq directly. So it shows it is related to the conc liq.

File: src/state/pwallet/concentratedLiquidity/hooks/evm.ts (hunk 1 above)
Comment (bilalcorbacioglu) [reply src/state/pwallet/concentratedLiquidity/hooks/evm.ts:118]: done

File: src/state/pwallet/concentratedLiquidity/types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (bilalcorbacioglu) [reply src/state/pwallet/concentratedLiquidity/types.ts:13]: I removed them directly. It seems we already used it at the bottom of that file.

**Pull request merged**